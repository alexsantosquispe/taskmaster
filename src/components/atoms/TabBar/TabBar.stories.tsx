import type { Meta, StoryObj } from '@storybook/react';
import TabBar from './TabBar';

const TABS = [
  {
    id: '1',
    label: 'Board'
  },
  {
    id: '2',
    label: 'To-Do'
  },
  {
    id: '3',
    label: 'Table'
  },
  {
    id: '4',
    label: 'List'
  }
];

const meta = {
  title: 'Atoms/TabBar',
  component: TabBar,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof TabBar>;

export default meta;

type Story = StoryObj<typeof TabBar>;

export const Default: Story = () => {
  return <TabBar tabs={TABS} />;
};

Default.args = {};
