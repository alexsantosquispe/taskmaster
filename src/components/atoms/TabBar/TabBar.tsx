import { useState } from 'react';
import { twMerge } from 'tailwind-merge';
import type { TabItem } from '../../../models/types';
import { TabBarItem } from './components/TabBarItem';

interface TabBarProps {
  tabs: TabItem[];
  className?: string;
}

const TabBar = ({ tabs, className }: TabBarProps) => {
  const [tabSelected, setTabSelected] = useState(tabs[0].id);

  return (
    <div
      className={twMerge(
        'flex w-full rounded-lg bg-neutral-100 p-0.5 text-sm md:w-fit dark:bg-neutral-800',
        className
      )}
    >
      {tabs.map((tab) => (
        <TabBarItem
          key={tab.id}
          tabItem={tab}
          isSelected={tabSelected === tab.id}
          onSelectTab={setTabSelected}
        />
      ))}
    </div>
  );
};

export default TabBar;
