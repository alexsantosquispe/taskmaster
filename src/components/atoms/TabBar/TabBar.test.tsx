import '@testing-library/jest-dom';

import { fireEvent, render, screen } from '@testing-library/react';

import TabBar from './TabBar';

const TABS = [
  {
    id: '1',
    label: 'Board'
  },
  {
    id: '2',
    label: 'To-Do'
  }
];

describe('TabBar', () => {
  describe('styles', () => {
    it('should render the component correctly', () => {
      const component = render(<TabBar tabs={TABS} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    const selectedTabStyles =
      'dark:bg-primary text-primary bg-white shadow dark:text-white';
    it('should change between tabs', () => {
      render(<TabBar tabs={TABS} />);

      const tabs = screen.getAllByRole('button');
      expect(tabs.length).toBe(TABS.length);

      expect(tabs[0]).toHaveClass(selectedTabStyles);
      expect(tabs[1]).not.toHaveClass(selectedTabStyles);

      fireEvent.click(tabs[1]);

      expect(tabs[0]).not.toHaveClass(selectedTabStyles);
      expect(tabs[1]).toHaveClass(selectedTabStyles);
    });
  });
});
