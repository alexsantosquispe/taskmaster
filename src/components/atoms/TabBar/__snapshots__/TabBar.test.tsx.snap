// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TabBar styles should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex w-full rounded-lg bg-neutral-100 p-0.5 text-sm md:w-fit dark:bg-neutral-800"
      >
        <button
          aria-label="Board"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md px-4 font-semibold transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 dark:bg-primary text-primary bg-white shadow dark:text-white"
        >
          <span>
            Board
          </span>
        </button>
        <button
          aria-label="To-Do"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md px-4 font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 dark:text-white/70"
        >
          <span>
            To-Do
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex w-full rounded-lg bg-neutral-100 p-0.5 text-sm md:w-fit dark:bg-neutral-800"
    >
      <button
        aria-label="Board"
        class="flex h-7 w-full items-center justify-center gap-1 rounded-md px-4 font-semibold transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 dark:bg-primary text-primary bg-white shadow dark:text-white"
      >
        <span>
          Board
        </span>
      </button>
      <button
        aria-label="To-Do"
        class="flex h-7 w-full items-center justify-center gap-1 rounded-md px-4 font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 dark:text-white/70"
      >
        <span>
          To-Do
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
